app/controllers/application_controller.rb

```diff
@@ -1,9 +1,17 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
 
-  def current_user
-    @current_user ||= User.first
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
   end
 
-  helper_method :current_user
+  def configure_devise_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |user|
+      user.permit(:name, :email, :password, :password_confirmation)
+    end
+  end
```

Comment:
- [ ] **[OPTIONAL]** It would be a good idea to prevent from visiting pages if they are not signed up, by authenticating the application controller.

  ```
  before_action :authenticate_user!
  ```